Mathlib/Analysis/BoxIntegral/UnitPartition.lean

```diff
@@ -293,4 +308,183 @@ theorem prepartition_isPartition {B : Box ι} (hB : hasIntegralVertices B) :
   rw [TaggedPrepartition.mem_toPrepartition, mem_prepartition_iff]
   exact ⟨index n x, mem_admissibleIndex_of_mem_box n hB hx, rfl⟩
 
+theorem exists_admissibleIndex {B : Box ι} (hB : hasIntegralVertices B) {x : ι → ℝ} (hx : x ∈ B) :
+    ∃ ν ∈ admissibleIndex n B, box n ν = box n (index n x) :=
+  ⟨index n x, mem_admissibleIndex_of_mem_box n hB hx, rfl⟩
+
+open Submodule Pointwise BigOperators
+
+open scoped Pointwise
+
+variable (c : ℝ) (s : Set (ι → ℝ)) (F : (ι → ℝ) → ℝ)
+
+local notation "L" => span ℤ (Set.range (Pi.basisFun ℝ ι))
+
+theorem tag_mem_smul_span (ν : ι → ℤ) :
```

Comment:
Maybe have a general lemma
```lean
lemma mem_smul_span_iff (v : ι → ℝ) : v ∈ (n : ℝ)⁻¹ • L ↔ ∀ i, n * v i ∈ Set.range ⇑(algebraMap ℤ ℝ)
```
and use it here and below?

Reply:
Yes, that's a good idea.